pkg/dnsmanager/dns_manager_test.go

```diff
@@ -38,30 +39,51 @@ func TestResolveIPAddress(t *testing.T) {
 			want: "",
 		},
 		{
-			name:   "dns resolution fallback",
+			name:   "no address",
 			ipAddr: "67.225.146.248",
 			dnsEvent: tracerdnstype.Event{
 				DNSName:    "test.com",
-				NumAnswers: 1,
+				NumAnswers: 0, // will not resolve
 			},
-			want: "test.com",
+			want: "",
 		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dm := &DNSManager{}
+			dm.ProcessDNSEvent(tt.dnsEvent)
+			got, _ := dm.ResolveIPAddress(tt.ipAddr)
+			if got != tt.want {
+				t.Errorf("ResolveIPAddress() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestResolveIPAddressFallback(t *testing.T) {
```

Comment:
@matthyx separated the fallback test case to a different test since IP address resolution has changed and it makes the test fail so I'm doing a `LookupIP` before